core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -241,7 +240,23 @@ class LogCleaner(val config: CleanerConfig,
           } finally {
             cleanerManager.doneCleaning(cleanable.topicPartition, cleanable.log.dir.getParentFile, endOffset)
           }
+          true
       }
+      val deleted = cleanerManager.logsReadyToBeTruncated() match {
```

Comment:
It seems it's simpler to just get the Iterable from logsReadyToBeTruncated() and call foreach on it, instead of doing a match?